### PR TITLE
docs: Refactor documentation

### PR DESCRIPTION
**Pull request by arthurpitman**
This PR refactors all documentation. It aims to:
- Simplify installation
- Be clear about how to set up a project
- Outline the key use-cases of the dynatrace-service
- Concentrate on SLI retrieval from YAML files and dashboards

It should lay the foundation for:
- #592
- #593
- #518 

Signed-off-by: Arthur Pitman <arthur.pitman@dynatrace.com>

### CODE REVIEW

**Review by j-poecher — DISMISSED** [review]
lgtm

File: documentation/problem-forwarding-to-keptn.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Forwarding problem notifications from Dynatrace to Keptn
+
+If you use Keptn to deploy your microservices and follow the standard tagging practices, Dynatrace will tag your monitored services with `keptn_project`, `keptn_service` and `keptn_stage`. If Dynatrace then detects a problem in one of these deployed services, e.g: High Failure Rate, Slow response time, you can let Dynatrace send these problems back to Keptn. To allow a Dynatrace problem to trigger a remediation workflow in Keptn, the dynatrace-service can intercept `sh.keptn.events.problem` events and forward the payload as a `sh.keptn.event.problem.open` event to the correct Keptn project, service and stage. All relevant problem details such as `PID`, `ProblemTitle` and `ProblemURL`are included.
```
Comment (johannes-b) [inline documentation/problem-forwarding-to-keptn.md:3]: *and forward the payload as a `sh.keptn.event.problem.open` event to the correct Keptn project, service and stage.* 

This fact is not correct anymore. Please update it to: 

*and triggers a `remediation` sequence by sending a `sh.keptn.event.[STAGE_NAME].remediation.triggered` event for the correct Keptn project, service, and stage.*

File: documentation/problem-forwarding-to-keptn.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        "Tags":"{Tags}",
+        "ImpactedEntities":{ImpactedEntities},
+        "ImpactedEntity":"{ImpactedEntity}",
+    }
+}
+```
+
+The dynatrace-service can [configure this automatically in a Dynatrace tenant](auto-tenant-configuration.md#problem-notifications).
+
+The dynatrace-service will parse the `Tags` field and use the `keptn_project`, `keptn_service` and `keptn_stage` tags that come directly from the impacted entities to ensure that the event is mapped to the correct project service and stage.
```
Comment (johannes-b) [inline documentation/problem-forwarding-to-keptn.md:35]: add a comma between: "project, service, and stage"

**Review by agrimmer — APPROVED** [review]
LGTM

File: documentation/problem-forwarding-to-keptn.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        "ImpactedEntities":{ImpactedEntities},
+        "ImpactedEntity":"{ImpactedEntity}",
+        "KeptnProject" : "dynatrace",
+        "KeptnService" : "allproblems",
+        "KeptnStage" : "production"
+    }
+}
+```
+
+When the dynatrace-service receives this `sh.keptn.events.problem` event it will parse the fields `KeptnProject`, `KeptnService` and `KeptnStage` and will then send a `sh.keptn.event.problem.open` event to Keptn including the rest of the problem details. This allows you to send any type of Dynatrace detected problem to Keptn and let Keptn execute a remediation workflow.
```
Comment (johannes-b) [inline documentation/problem-forwarding-to-keptn.md:70]: ... will then send a `sh.keptn.event.[STAGE_NAME].remediation.triggered` event to Keptn.

File: documentation/problem-forwarding-to-keptn.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+*Best Practice:* Dynatrace alerting profiles can be used to filter certain problem types, e.g: infrastructure problems in production or slow Performance in developer environment. By creating a Keptn project to handle these remediation workflows and a Keptn service for each alerting profile, it is easy to define workflows for particular problem types. Furthermore, individual environment names such as `pre-prod` or `production` can be represented as stages within the project.
+
+Here is a screenshot of a workflow triggered by a Dynatrace problem and how it then executes in Keptn:
+
+![Workflow triggered by a Dynatrace problem](images/remediation_workflow.png "Workflow triggered by a Dynatrace problem")
```
Comment (johannes-b) [inline documentation/problem-forwarding-to-keptn.md:76]: Please update: images/remediation_workflow.png using the next png. 

image

File: documentation/auto-service-onboarding.md [hunk 5]
```diff
@@ -16,45 +15,23 @@ spec:
       test_strategy: "performance"
```
Comment (johannes-b) [inline documentation/auto-service-onboarding.md:15]: Please remove the line `test_strategy: "performance"`

**Comment by johannes-b** [comment]
Out of curiosity and because I could not read it, but do we still support the feature that updates a problem in Dynatrace when a remediation action is triggered/finished? 

This feature was available and the outcome looked as follows (see comment section): 

image

**Comment by johannes-b** [comment]
Other than the comments I left, I could not spot any further issues in this PR. Great work!

**Review by j-poecher — APPROVED** [review]
lgtm

**Pull request merged**